### PR TITLE
feat(zsh-plugins): add feature that installs zsh plugins

### PR DESCRIPTION
**Pull request by gund**
Hi, I would like to contribute a feature that I've recently built for my own dev container setup that allows to install (Oh-My-)ZSH plugins from configuration.

Here's the example of how it's intended to be used:
```jsonc
{
  "features": {
    "ghcr.io/devcontainers/features/common-utils:1": {},
    "./zsh-plugins": {
      "plugins": "npm zsh-autosuggestions", // <-- zsh plugins go here
      "omzPlugins": "https://github.com/zsh-users/zsh-autosuggestions" // <-- oh-my-zsh plugin git urls go here
    }
  }
}
```

I was trying to follow your contributing guide but it seems outdated as it says to run script from `scripts/` folder to generate a feature but there is no such folder anymore.
So I basically created a feature manually and added a test file which for now does not really test the plugin and I need help with it.
Basically I need a way to configure a feature inside a test to set some zsh plugins to be able to assert that they were installed.
Could you please help me how this may be achieved?

Off-topic: I have 2 more generic features that I would like to contribute: one for persisting bash/zsh history in volume so that container rebuilds won't loose commands history and the second feature for persisting vscode extensions folder to allow for faster startup times after container is rebuilt, so would it be fine to contribute them here as well?

Thanks!

### CODE REVIEW

**Comment by gund** [comment]
Oh I just discovered from another PR here that [scenarios](https://github.com/devcontainers/cli/blob/main/docs/features/test.md#scenarios) is probably what I need to use for my feature testing to setup zsh plugins?

**Comment by gund** [comment]
So I've tried to add a scenario to test the feature but unfortunately it fails to picking zsh plugins in the tests, so I'm not quite sure what I'm doing wrong in the tests but I'm trying to execute tests under `zsh -c "cmd"` which should have the plugins configured but they are not there for some reason.

Any clues what may be the problem?

**Comment by danielbraun89** [comment]
Looks good to me, Altho there are some voices for comma-separated values instead of spaces (https://github.com/devcontainers/spec/issues/57#issuecomment-1309015728) I would say thats a more of a matter of preference,  lets get it merged!
